app/services/curation_concerns/derivative_path.rb

```diff
@@ -1,49 +1,67 @@
 module CurationConcerns
   class DerivativePath
+    attr_reader :id, :destination_name
+
     class << self
       # Path on file system where derivative file is stored
+      # @param [ActiveFedora::Base or String] object either the AF object or its id
+      # @param [String] destintation_name
       def derivative_path_for_reference(object, destination_name)
-        destination_name = destination_name.gsub(/^original_file_/, '')
-        derivative_path(object, extension_for(destination_name), destination_name)
+        new(object, destination_name).derivative_path
       end
 
+      # @param [ActiveFedora::Base or String] object either the AF object or its id
       # @return [Array<String>] Array of paths to derivatives for this object.
       def derivatives_for_reference(object)
-        Dir.glob(root_path(object).join("*")).select do |path|
-          path.start_with?(path_prefix(object).to_s)
-        end
+        new(object).all_paths
       end
+    end
 
-      private
+    # @param [ActiveFedora::Base or String] object either the AF object or its id
```

Comment:
use a comma rather than 'or' to separate types. See http://yardoc.org/types.html